Cache/Main.lean

```diff
@@ -71,15 +73,25 @@ def main (args : List String) : IO Unit := do
     println "Unfortunately, you have a broken Lean v4.8.0-rc1 installation."
     println "Please run `elan toolchain uninstall leanprover/lean4:v4.8.0-rc1` and try again."
     Process.exit 1
-  -- We pass any following arguments to `getHashMemo`,
-  -- so we can use the cache on `Archive` or `Counterexamples`.
-  let extraRoots := match args with
-  | [] => #[]
-  | _ :: t => t.toArray.map FilePath.mk
   if args.isEmpty then
     println help
     Process.exit 0
   CacheM.run do
+
+  let mut roots : Std.HashMap Lean.Name FilePath ← parseArgs args
+  if roots.isEmpty then do
+    -- No arguments means to start from `Mathlib.lean`
+    -- TODO: could change this to the default-target of a downstream project
+    let mod := `Mathlib
+    let sp := (← read).srcSearchPath
+    let sourceFile ← Lean.findLean sp mod
+    roots := Std.HashMap.empty.insert mod sourceFile
```

Comment:
This is new, and could probably be moved to a follow-up. It has no relevance here since at the very bottom of `Cache/Hashing.lean`, `Mathlib.lean` is currently always added as a key.